storage/doc.go

```diff
@@ -274,15 +274,43 @@ To generate the signature, you must have:
 
 # Errors
 
-Errors returned by this client are often of the type [googleapi.Error].
-These errors can be introspected for more information by using [errors.As]
-with the richer [googleapi.Error] type. For example:
+Errors returned by this client are often of the type [googleapi.Error] or
+[github.com/googleapis/gax-go/v2/apierror.APIError].
+The [apierror.APIError] type can wrap a [google.golang.org/grpc/status.Status]
+if gRPC was used, or a [google.golang.org/api/googleapi.Error] if HTTP/REST was used.
+You might also encounter `[googleapi.Error]` directly from HTTP operations.
+These two types of errors can be inspected for more information by using [errors.As]
+to access the specific underlying error types and retrieve detailed
+information, including HTTP or gRPC status codes. For example:
 
 	var e *googleapi.Error
 	if ok := errors.As(err, &e); ok {
-		  if e.Code == 409 { ... }
+		// e.Code is the HTTP status code.
+		// e.Message is the error message.
+		// e.Body is the raw response body.
+		// e.Header contains the HTTP response headers.
+		log.Printf("HTTP Code: %d, Message: %s", e.Code, e.Message)
+
+		if e.HTTPCode() == 409 {
+			// ... handle conflict ...
+		}
 	}
 
+	var ae *apierror.APIError
+	if ok := errors.As(err, &ae); ok {
+		// ae can wrap a googleapi.Error or a status.Status (from gRPC)
+		log.Printf("APIError: HTTPCode: %d, GRPCStatusCode: %s", ae.HTTPCode(), ae.GRPCStatus().Code())
+
+		// To get the underlying gRPC status directly:
+		if s, ok := status.FromError(ae.Unwrap()); ok {
+			if s.Code() == codes.Unavailable {
+				// ... handle gRPC unavailable ...
+			}
+		}
+	}
+
+See GRPC Status Codes at https://pkg.go.dev/google.golang.org/grpc/codes.
```

Comment:
Thinking about it again, I don't think we need this link; you can find this on several of the functions in the example code.